dotCMS/src/main/java/com/dotcms/cms/login/LoginServiceAPIImpl.java

```diff
@@ -0,0 +1,537 @@
+package com.dotcms.cms.login;
+
+import static com.dotmarketing.util.CookieUtil.createJsonWebTokenCookie;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.CopyOnWriteArrayList;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import com.dotcms.api.web.HttpServletRequestThreadLocal;
+import com.dotcms.auth.providers.jwt.JsonWebTokenUtils;
+import com.dotcms.cms.login.events.LDAPLoginEventListener;
+import com.dotcms.cms.login.events.LoginEvent;
+import com.dotcms.cms.login.events.LoginEventListener;
+import com.dotcms.cms.login.events.LoginEvents;
+import com.dotcms.enterprise.BaseAuthenticator;
+import com.dotcms.enterprise.LDAPImpl;
+import com.dotcms.enterprise.LicenseUtil;
+import com.dotcms.enterprise.PasswordFactoryProxy;
+import com.dotcms.enterprise.cas.CASAuthUtils;
+import com.dotcms.enterprise.de.qaware.heimdall.PasswordException;
+import com.dotcms.mock.request.MockHttpRequest;
+import com.dotcms.mock.request.MockSessionRequest;
+import com.dotcms.mock.response.BaseResponse;
+import com.dotcms.mock.response.MockHttpResponse;
+import com.dotcms.repackage.com.google.common.annotations.VisibleForTesting;
+import com.dotcms.util.ReflectionUtils;
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.business.ApiProvider;
+import com.dotmarketing.business.DuplicateUserException;
+import com.dotmarketing.business.UserAPI;
+import com.dotmarketing.business.web.UserWebAPI;
+import com.dotmarketing.cms.factories.PublicCompanyFactory;
+import com.dotmarketing.cms.factories.PublicEncryptionFactory;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotSecurityException;
+import com.dotmarketing.portal.struts.DotCustomLoginPostAction;
+import com.dotmarketing.util.Config;
+import com.dotmarketing.util.Logger;
+import com.dotmarketing.util.SecurityLogger;
+import com.dotmarketing.util.UtilMethods;
+import com.liferay.portal.NoSuchUserException;
+import com.liferay.portal.PortalException;
+import com.liferay.portal.SystemException;
+import com.liferay.portal.auth.AuthException;
+import com.liferay.portal.auth.Authenticator;
+import com.liferay.portal.model.Company;
+import com.liferay.portal.model.User;
+import com.liferay.portal.util.PropsUtil;
+import com.liferay.portal.util.WebKeys;
+import com.liferay.util.Validator;
+
+/**
+ * Default implementation
+ */
+final class LoginServiceAPIImpl implements LoginServiceAPI {
+
+
+    private static final long serialVersionUID = 1L;
+    private final Log log = LogFactory.getLog(LoginServiceAPI.class);
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Make "log" transient or serializable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1948)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Make the "log" logger private static final and rename it to comply with the format "LOG(?:GER)?". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1312)